examples/diffusion/diffusion.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "llama.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef bool (*diffusion_step_callback_t)(int32_t step, int32_t total_steps, const llama_token * tokens,
+                                          int32_t n_tokens, void * user_data);
+
+enum diffusion_algorithm {
+    DIFFUSION_ALG_ORIGIN       = 0,
+    DIFFUSION_ALG_MASKGIT_PLUS = 1,
+    DIFFUSION_ALG_TOPK_MARGIN  = 2,
+    DIFFUSION_ALG_ENTROPY      = 3,
+};
```

Comment:
The convention is to match the enum name with the prefix of the values:

```suggestion
enum diffusion_alg {
    DIFFUSION_ALG_ORIGIN       = 0,
    DIFFUSION_ALG_MASKGIT_PLUS = 1,
    DIFFUSION_ALG_TOPK_MARGIN  = 2,
    DIFFUSION_ALG_ENTROPY      = 3,
};
```